services/port_manager/src/main/java/com/futurewei/alcor/portmanager/service/implement/PortServiceImpl.java

```diff
@@ -267,84 +372,70 @@ public PortStateJson updatePortState(String projectId, String portId, PortStateJ
         AsyncExecutor executor = new AsyncExecutor();
 
         PortState portState = portStateJson.getPortState();
-        PortState oldPortState = portRepository.findItem(portId);
+        portState.setProjectId(projectId);
 
         try {
-            if (portRepository.findItem(portId) == null) {
+            PortState oldPortState = portRepository.findItem(portId);
+            if (oldPortState == null) {
                 throw new PortStateNotFoundException();
             }
 
-            portState.setProjectId(projectId);
-
-            //Update mac_address
+            updatePortStateAsync(portState, oldPortState, executor, rollbacks);
```

Comment:
The update logic is quite tricky. Do we have sufficient UTs to cover those corner cases?